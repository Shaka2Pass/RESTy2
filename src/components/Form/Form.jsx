import React from 'react';
import { useResty } from '../../hooks/RestyProvider';


const Form = () => {
  const { url, method, body, onChange, onSubmit } = useResty();

  <section>
    <form className="Form" onSubmit={onSubmit} >
      <input type='text' name='url' value={url} 
        onChange={onChange}/>

      <input type='radio' name='method' value='GET' 
        checked={method === 'GET'} 
        onChange={onChange}/>
      <label>GET:</label>

      <input type='radio' name='method' value='POST' 
        checked={method === 'POST'} 
        onChange={onChange}/>
      <label>POST:</label>

      <input type='radio' name='method' value='PUT' 
        checked={method === 'PUT'} 
        onChange={onChange}/>
      <label>PUT:</label>

      <input type='radio' name='method' value='PATCH' 
        checked={method === 'PATCH'} 
        onChange={onChange}/>
      <label>PATCH:</label>

      <input type='radio' name='method' value='DELETE' 
        checked={method === 'DELETE'} 
        onChange={onChange}/>
      <label>DELETE:</label>

      <button>GO!</button>
      
      <textarea name="body" value={body} 
        onChange={onChange}>
      </textarea>

    </form>
  </section>;
};

export default Form;




