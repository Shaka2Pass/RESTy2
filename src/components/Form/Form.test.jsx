import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form.jsx';

describe('Form component', () => {
  it('matches a snapshot of Form.jsx', () => {
    const wrapper = shallow(<Form
      url="https://xfiles-api.herokuapp.com/api/v1/characters"
      method="GET"
      body="{}"
      onSubmit={()=>{}}
      onChange={()=>{}}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});

