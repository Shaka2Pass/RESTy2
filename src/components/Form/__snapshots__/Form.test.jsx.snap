// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component matches a snapshot of Form.jsx 1`] = `
<section>
  <form
    className="Form"
    onSubmit={[Function]}
  >
    <input
      name="url"
      onChange={[Function]}
      type="text"
      value="https://xfiles-api.herokuapp.com/api/v1/characters"
    />
    <input
      checked={true}
      name="method"
      onChange={[Function]}
      type="radio"
      value="GET"
    />
    <label>
      GET:
    </label>
    <input
      checked={false}
      name="method"
      onChange={[Function]}
      type="radio"
      value="POST"
    />
    <label>
      POST:
    </label>
    <input
      checked={false}
      name="method"
      onChange={[Function]}
      type="radio"
      value="PUT"
    />
    <label>
      PUT:
    </label>
    <input
      checked={false}
      name="method"
      onChange={[Function]}
      type="radio"
      value="PATCH"
    />
    <label>
      PATCH:
    </label>
    <input
      checked={false}
      name="method"
      onChange={[Function]}
      type="radio"
      value="DELETE"
    />
    <label>
      DELETE:
    </label>
    <button>
      GO!
    </button>
    <textarea
      name="body"
      onChange={[Function]}
      value="{}"
    />
  </form>
</section>
`;
